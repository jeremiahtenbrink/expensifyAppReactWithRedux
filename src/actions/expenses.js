import uuid from "uuid";

//ADD EXPENSE ACTION
export const addExpense = (
	{
		description = "",
		note = "",
		amount = 0,
		createdAt = 0,
	} = {} ) => ( {
	type   : "ADD EXPENSE",
	expense: {
		id: uuid(),
		description,
		note,
		amount,
		createdAt,
	},
	
} );

//EDIT EXPENSE ACTION
export const editExpense = (
	{
		id,
		updates,
	} = {} ) => ( {
	type: "EDIT EXPENSE",
	id,
	updates,
} );

// REMOVE EXPENSE BY ID
export const removeExpense = ( { id = undefined } = {} ) => ( {
	type: "REMOVE EXPENSE",
	id,
} );
