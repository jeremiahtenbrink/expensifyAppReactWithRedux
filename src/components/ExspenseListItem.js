import React from "react";
import { Link } from "react-router-dom";

const ExspenseListItem = ( { description, amount, createdAt, id } ) => {
	return (
		<div>
			<h3><Link to={ "/edit/" + id }>{ description }</Link></h3>
			<p>{ amount } -- { createdAt }</p>
		
		</div>
	);
};

export default ExspenseListItem;
ExspenseListItem.propTypes = {};
ExspenseListItem.defaultProps = {};

