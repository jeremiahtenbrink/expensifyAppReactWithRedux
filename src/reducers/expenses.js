const expensesReducerDefaultState = [];

export default ( state = expensesReducerDefaultState, action ) => {
	switch ( action.type ){
		case "ADD EXPENSE":
			return [
				...state,
				action.expense,
			];
		case "REMOVE EXPENSE":
			return state.filter( ( expense ) => expense.id !== action.id );
		case "EDIT EXPENSE":
			return state.map( ( expense ) => {
				if ( expense.id === action.id ){
					const newExpense = { ...expense, ...action.updates };
					
					return newExpense;
					
				} else{
					return expense;
				}
			} );
		default:
			return state;
	}
};

