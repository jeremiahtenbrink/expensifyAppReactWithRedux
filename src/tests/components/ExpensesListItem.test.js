import React from "react";
import ExspenseListItem from "../../components/ExspenseListItem";
import { shallow } from "enzyme/build";
import expenses from "../fixtures/expenses";

test('should render expense list item with given item', () => {
	const wrapper = shallow( <ExspenseListItem key={ expenses[1].id } { ...expenses[1] } /> );
	expect(wrapper).toMatchSnapshot();
});