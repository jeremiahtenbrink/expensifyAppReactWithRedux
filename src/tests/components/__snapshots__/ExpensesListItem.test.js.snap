// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item with given item 1`] = `
<div>
  <h3>
    <Link
      replace={false}
      to="/edit/2"
    >
      Rent
    </Link>
  </h3>
  <p>
    109500
     -- 
    -345600000
  </p>
</div>
`;
